python/ray/_private/worker.py

```diff
@@ -2343,29 +2343,38 @@ def connect(
 
     if mode == SCRIPT_MODE:
```

Comment:
Btw, this code path is added only when the process is a driver. What's the reason why it is set for worker processes?